client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -116,7 +116,12 @@ const CheckoutThankYou = React.createClass( {
 		if ( this.props.receipt.hasLoadedFromServer && getPurchases( this.props ).every( isTheme ) ) {
 			this.props.activatedTheme( getPurchases( this.props )[ 0 ].meta, this.props.selectedSite );
 
-			page.redirect( '/design/' + this.props.selectedSite.slug );
+			// If we're coming from the signup funnel, redirect to the frontend
+			if ( getPurchases( this.props )[0].source === 'signup-with-theme' ) {
```

Comment:
This check doesn't seem to be working. Dump of the purchase object:
<img width="380" alt="screen shot 2016-03-30 at 14 59 14" src="https://cloud.githubusercontent.com/assets/7767559/14144575/2a1940de-f688-11e5-8794-d013f5548f11.png">